DesignDoc/RMA.md

```diff
@@ -1,3 +1,205 @@
+   * site information where the defective parts resides
+   * contact information from Consumer side
+   * RMA service level, e.g., normal / escalate service
+
+1. Consumer request for RMA from Publisher
+1. Request type for each customer (email, API, manual).
+
+### Review/Approval 
+
+At this stage, Publisher will validate the required information for the RMA case, and approve / reject / request more information, which may include:
+   * whether any mandotary information is missing
+   * whether a kind of service level is supported for specific region
+
+### Parts tracking 
+
+There are two seperate flows at this stage:
+   1. Replacement shippment:
+      1. Publisher will provide the shippment information, including carrier, planned arrive date 
+      1. Publisher will notify the status of the replacement shippment and delivery
+   2. Defective part shippment:
+      1. Consumer will provide the shippment information, including carrier, planned arrive date 
+      1. Publisher will notify the status of the replacement shippment and delivery
+
+### Notification
+
+One RMA case is dedicated to one part with one unique identifier, e.g., serial number. Each RMA involves multiple communication messages between the publisher and consumer. Each RMA case has a unique ID.
+
+## Notification structure
+
+### General format
+
+`field_name:field_type:description`
+
+### Field Details
+
+#### case_id
+
+Identifier for a RMA case which dedicated to a part with a unique identifier. 
+
+`case_id:string:"RMA case identifier"`
+
+
+#### case_status
+
+RMA case status 
+`case_status:enum value:"stage of RMA case" enumeration: [OPEN, APPROVED, REJECTED]`
+
+
+#### case_status_time
+
+Epoch time when the status is updated from Publisher, we have one specific time for each status listed in `case_status`
+
+`case_open_time:date time:"Unix epoch time when consumer starts to request the RMA"`
+`case_approved_time:date time:"Unix epoch time when publisher approve to request the RMA"`
+`case_rejected_time:date time:"Unix epoch time when publisher reject the RMA"`
+
+#### part_shippment_status 
+
+Replacement / defective part shippment status
+
+`service_type:enum value:"Type of service: leased wave/dark fiber" enumeration: [LEASED_WAVE, DARK_FIBER]`
+
+#### part_type
+
+Replacement or defective parts
+
+#### service_impact
+
+Type of impact caused on the service because of the planned work.
+
+`service_impact:enum value:"Type of service impact." enumerations: [COMPLETE_OUTAGE, POSSIBLE_OUTAGE, INTERMITTENT_OUTAGE]`
+
+### impact_type
+
+Type of impact for the maintenance. If the outage would be complete, intermittent or possible.
+
+`impact_type:enum value:"If this impact is primary or secondary for a planned maintenance." enumerations: ['COMPLETE_OUTAGE', 'POSSIBLE_OUTAGE', 'INTERMITTENT_OUTAGE']`
+
+### impact_satus
+
+Status of the work.
+
+`impact_satus:enum value:"Status of the impact." 
+enumerations: [SCHEDULED, IN_PROGRESS, COMPLETED, CANCELED]
+`
+
```

Comment:
All the fields mentioned from 119-185 this line are for fiber maintenance, they should be removed from RMA notification structure.